File: tests/.hooks/csharp/primary/src/Openapi/Hooks/HookRegistration.cs
namespace Openapi.Hooks
{
    public static class HookRegistration
    {
        public static void InitHooks(IHooks hooks)
        {
            var testHook = new TestHook();
            hooks.RegisterSDKInitHook(testHook);
            hooks.RegisterBeforeRequestHook(testHook);
            hooks.RegisterAfterSuccessHook(testHook);
            hooks.RegisterAfterErrorHook(testHook);

            var customSecurityHook = new CustomSecurityHook();
            hooks.RegisterBeforeRequestHook(customSecurityHook);

            var testCancellationHook = new TestCancellationHook();
            hooks.RegisterBeforeRequestHook(testCancellationHook);
        }
    }
}

File: tests/.hooks/php/primary/src/Hooks/HookRegistration.php
<?php

declare(strict_types=1);

namespace OpenAPI\OpenAPI\Hooks;

class HookRegistration
{
    /**
     * @param  Hooks  $hooks
     */
    public static function initHooks(Hooks $hooks): void
    {
        $testHook = new TestHook();
        $hooks->registerSDKInitHook($testHook);
        $hooks->registerBeforeRequestHook($testHook);
        $hooks->registerAfterSuccessHook($testHook);
        $hooks->registerAfterErrorHook($testHook);

        $customSecurityHook = new CustomSecurityHook();
        $hooks->registerBeforeRequestHook($customSecurityHook);
    }
}
